### PR TITLE
Document SOFTMAX in the activations strategy table (Issue #3690)

### LINKED ISSUE #3690: 🟡 Activations README strategy table omits SOFTMAX — 38 of 39 registered squashes documented
`src/methods/activations/README.md:58-97` — the backpropagation-strategy table covers 38 activation functions, but the central registry (`src/methods/activations/Activations.ts:201-256`) registers 39. The missing entry is `SOFTMAX` (imported at `Activations.ts:44` from `src/methods/activations/types/SOFTMAX.ts`): it appears nowhere in the README — no priority weight, no invertibility note, no derivative-vs-unSquash recommendation. `CONTRIBUTING.md:471` calls this README "the full reference" for activation functions, and `CONTRIBUTING.md:498-501` requires every new activation to add an entry "following the existing format — include priority, invertibility, and backpropagation strategy" (checks 2 and 6, `severity:medium`).

## Why this matters

The table presents itself as the complete per-squash strategy reference, so a reader reasonably concludes `SOFTMAX` does not exist as a squash — despite it being registered for mutation selection and load-bearing in practice (it is the output activation the `CROSS_ENTROPY` cost coupling selects, used by downstream factory-seeded classifiers such as the MNIST example in NEAT-AI-Examples). A contributor auditing priorities or backprop strategies from this table silently skips one real function, and the documented add-an-entry rule has already been violated once, which invites further drift.

## Suggested fix

Add a `SOFTMAX` row to the table in `src/methods/activations/README.md` following the existing format — its priority weight, invertibility (softmax is not element-wise invertible), and the recommended error-propagation strategy (derivative-based via the softmax Jacobian / cross-entropy shortcut). Consider also adding a one-line note that the registry in `Activations.ts` is the authoritative list, so the next mismatch is caught at review time.

### PR DESCRIPTION
**Pull request by VibeCoderST**
# PR Summary — Issue #3690

## Summary

The backpropagation-strategy table in `src/methods/activations/README.md`
documented 38 activations while `src/methods/activations/Activations.ts`
registers 39 — `SOFTMAX` appeared nowhere, despite being the output activation
the `CROSS_ENTROPY` coupling selects. Closes #3690.

Changes:

- Added the `SOFTMAX` row to the strategy table, plus a note explaining why it
  is the odd one out: true softmax is a vector operation, so `unSquash()`
  inverts the per-neuron logistic surrogate rather than the normalised vector
  (`softmaxNormalise()` does the vector form); `calculateError()` returns the
  softmax + cross-entropy gradient (`target − activation`) with no derivative
  scaling; and its `⬇ 0` priority means "chosen explicitly on an output layer",
  not "deprecated or unsuitable" like the other zero-priority rows.
- Extended the Invertible legend with `⚠️` (surrogate-only inversion) so the new
  row's marker is defined.
- Added an `[!IMPORTANT]` note naming `Activations.ts` as the authoritative list
  that the table documents.
- `CONTRIBUTING.md` step 4 now points at the guard test.

## Evidence

Documentation change with no web interface to screenshot. The evidence is the
new guard test, which fails against the unfixed README and passes after it.

Before the README fix:

```text
activations README documents every registered squash ... FAILED
  [Diff] Actual / Expected
  -   [ "SOFTMAX" ]
  +   []
activations README names the registry as authoritative ... FAILED
FAILED | 1 passed | 2 failed
```

After:

```text
activations README documents every registered squash ... ok
activations README has no rows for unregistered squashes ... ok
activations README names the registry as authoritative ... ok
ok | 3 passed | 0 failed
```

Full gate: `./quality.sh < /dev/null` →
`ok | 8222 passed (5 steps) | 0 failed |
4 ignored (1m11s)`.

```mermaid
flowchart LR
    A["Activations.ts registry<br/>39 squashes"] -->|authoritative| B["README strategy table"]
    A --> C["test/docs/<br/>ActivationStrategyTable.ts"]
    B --> C
    C -->|"missing row / orphan row"| D["❌ quality gate fails"]
    C -->|"in sync"| E["✅ passes"]
```

## Test Plan

- Added `test/docs/ActivationStrategyTable.ts`:
  - `activations README documents every registered squash` — every canonical
    name from `Activations.list()` has a table row (regression test for the
    missing `SOFTMAX` entry).
  - `activations README has no rows for unregistered squashes` — the reverse
    direction, so a removed or renamed squash cannot leave a stale row.
  - `activations README names the registry as authoritative` — the README points
    readers at `Activations.ts`.
- No existing tests modified or removed.

### CODE REVIEW

**Comment by VibeCoderST** [comment]
✅ **Vibe Coder released this claim** — host `GRQ-23`, finished 17:17 UTC.

**Progress**
- 17:11 Claimed PR #3721 (CI check `markdownlint` failing)
- 17:12 Diagnosing `markdownlint` (code-fix-required) — fix attempt 1 of 3
- 17:17 No code changes produced for `markdownlint`
- 17:17 Released — Reviewed CI failure for PR #3721 (markdownlint) — no changes needed

**Comment by VibeCoderST** [comment]
## CI failure needs human attention

**Why:** I reviewed the CI check failure (**markdownlint**) but could not produce a code fix. The classifier indicates this requires a code change.

**Classifier reason:** markdownlint finding
**Signals:**
- `check:markdownlint`

**Next step:** Review the failing check and either push a fix or reply with guidance, then remove the `needs-human` label so the worker can retry.

---
🤖 Processed by: the worker
